### PR TITLE
setup github actions

### PR DESCRIPTION
**Pull request by yoshuawuyts**
The migration to the http-rs org disabled CI. This sets up github actions for use here. Thanks!

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      uses: actions-rs/cargo@v1
+      with:
+        command: check
+        args: --all --bins --examples
+
+    - name: check unstable
+      uses: actions-rs/cargo@v1
+      with:
+        command:  check
+        args: --all --benches --bins --examples --tests
```
Comment (k-nasa) [inline .github/workflows/ci.yaml]: Is Coco necessary?
I don't think there is a need to separate 'check' from 'check unstable'.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
```
Comment (k-nasa) [inline .github/workflows/ci.yaml]: Do you need master?

Comment (yoshuawuyts) [reply .github/workflows/ci.yaml]: For Bors the "trying" branch is usually enough. Perhaps we should add master too though.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (yoshuawuyts) [reply .github/workflows/ci.yaml]: Oh yes, great point. I copied this in accidentally. Thanks heaps!

**Pull request merged**